components/logins/src/logins.udl

```diff
@@ -72,15 +68,6 @@ dictionary Login {
     SecureLoginFields sec_fields;
 };
 
-/// An encrypted version of [Login].  This is what we return for all the "read"
-/// APIs - we never return the cleartext of encrypted fields.
-dictionary EncryptedLogin {
```

Comment:
I'm really happy to see this die. Back in the day, we were torn about whether the logins store should be able to function while in a "locked" state - ie, we were picturing a world where the app knows a login exists for a site, but the app must force the user to unlock via biometrics etc before the actual details were available, which then allows the app to "re-lock" the store at its convenience.

It looks like you've decided this is a non-goal, and your team is the correct team to make that decision, and the extra simplicity that comes with that is a nice bonus!

You might be able to take this further though - it seems that without `EncryptedLogin` you might also be able to flatten out  the `Login` record so avoid the nested records (ie, can RecordFields, and SecureLoginFields now die?) While that might be able to be done as a followup, the cost of re-breaking the mobile platforms might mean the opportunity to do so might not arise. IMO, the current splitting of these records makes the use of this component in the mobile code quite complicated to understand so would likely benefit from a simplification here.

(alternatively, a different split might make sense - eg, split out "editable" fields from "computed" fields, but even that might not have enough value)

Reply:
Yes, I think the advantages of a simplified way of working for LoginsStore outweigh the disadvantages. The idea of a locked state further complicates things and has a knock-on effect at lower levels. We can still perform certain operations without the store unlocked – for example, I added (thanks @bendk!) the method `has_logins_by_base_domain(&self, base_domain: &str)` so that the application can still check whether logins exist for a specific domain.

I like that you suggest further simplifying the login structure. I hadn't even dared to tackle that :) I'll discuss it with the mobile teams. On Android, the code is already abstracted, so the changes would be very simple here. On iOS, I'm not sure. At the latest now, a decoupling on the iOS side seems sensible to me in order to be able to make adjustments more agilely.

Furthermore, we would actually like to encrypt more fields, especially the domain. This information could already be interesting for certain attackers and some users could therefore support such changes. Although this would complicate or even prevent the lookup by domain in the locked state.

Reply:
Yeah! I'm a little torn by `RecordFields` though - roughly speaking, they are fields which the consumer should never *provide* - eg, when creating a login you never specify the `id` or the number of times it has been used. This is a tension we often face - use different structs for "get" vs "set" operations, which makes consumers a little more painful in some ways, but makes things clearer in others.

But certainly the flattening of `LoginFields` looks like a big improvement.

Reply:
Yes, I know, this is an old problem. And then, of course, there is something appealing about grouping all these meta fields, which only play a role when reading, into a separate _record_ struct. And on the other hand, there is this simplicity and clarity ¯\_(ツ)_/¯

Reply:
exactly :) It sounds like we agree there's no objectively correct answer here, so I can get behind whatever you decide!

Reply:
So, I've implemented this here. It makes perfect sense to me, it will be much easier and cleaner for consumers, which can be nicely seen from the sync-pass example. Thanks again for intervening here!